resource/traversers/dfu.cpp

```diff
@@ -77,6 +76,58 @@ int dfu_traverser_t::schedule (Jobspec::Jobspec &jobspec,
     planner_multi_t *p = NULL;
     const subsystem_t &dom = get_match_cb ()->dom_subsystem ();
 
+    // precheck to see if enough resources are available for this to be feasible
+    if (op != match_op_t::MATCH_UNKNOWN) {
+        bool checking_satisfiability = op == match_op_t::MATCH_ALLOCATE_W_SATISFIABILITY || op == match_op_t::MATCH_SATISFIABILITY;
+        graph_duration_t graph_duration = get_graph_db ()->metadata.graph_duration;
+        auto &g = *get_graph ();
+        int64_t graph_end = std::chrono::duration_cast<std::chrono::seconds>
+            (graph_duration.graph_end.time_since_epoch ()).count ();
+        auto p_iter = g[root].idata.subplans.find (dom);
+        auto by_type_iter = get_graph_db ()->metadata.by_type.find ("node");
+        int64_t target_time = graph_end - 1;
+        int64_t target_duration = 1;
```

Comment:
Considering the requested duration might be better:
```suggestion
        int64_t target_time = graph_end - meta.duration;
        int64_t target_duration = meta.duration;
```

Reply:
Thanks for this, I had tried using the same as the schedule function uses below (graph_end - meta.duration - 1) and it didn't pass, but this does.  Changed.

Reply:
I have to amend this a bit.  It has to use an at of `graph_end - 1` for cases other than ALLOCATE because duration may actually be the whole graph, except the literal last second.  We can use the meta.duration directly in all cases though so I'm now doing that.